test/serializers/associations_test.rb

```diff
@@ -207,6 +207,49 @@ def test_associations_namespaced_resources
           end
         end
       end
+
+      class NamespaceNestedSerializersTest < Minitest::Test
```

Comment:
This snippet from the PR description would also make a great test showing the old behavior and the behavior this PR introduces.  It wasn't obvious to me from the examples why you'd want to 'step back' one level in the serializer lookup.

> It simply modifies Serializer#serializer_lookup_chain_for to add all
of the possible levels of namespaces based on the current serializer.
> With this patch for example if we have:

> class Public::V1::PostSerializer
  belongs_to :author
end

> class Public::V1::AuthorSerializer
> end
> we will wind up with the following items in the lookup chain for an
author when serializing a Post with Public::V1::PostSerializer:

> ['Public::V1::PostSerializer::AuthorSerializer',
'Public::V1::AuthorSerializer', 'Public::AuthorSerializer',
'::AuthorSerializer']